cpp/example_code/s3/put_object_acl.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+#include <iostream>
+#include <aws/core/Aws.h>
+#include <aws/s3/model/Permission.h>
+#include <aws/s3/model/Type.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/Owner.h>
+#include <aws/s3/model/Grantee.h>
+#include <aws/s3/model/Grant.h>
+#include <aws/s3/model/AccessControlPolicy.h>
+#include <aws/s3/model/GetObjectAclRequest.h>
+#include <aws/s3/model/PutObjectAclRequest.h>
+#include <awsdoc/s3/s3_examples.h>
+
+/*
+ * Before running this C++ code example, set up your development environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/AmazonS3/latest/userguide/GetStartedWithS3.html
+ *
+ * Purpose
+ *
+ * Demonstrates using the AWS SDK for C++ to setting the access control list (ACL) for an object
+ * in an S3 bucket.
+ *
+ */
+
+static Aws::S3::Model::Permission SetGranteePermission(const Aws::String &access);
+
+static Aws::S3::Model::Type SetGranteeType(const Aws::String &type);
+
+//! Routine which demonstrates setting the ACL for an object in an S3 bucket.
+/*!
+  \sa PutObjectAcl()
+  \param bucketName Name of from bucket.
+  \param objectKey Name of object in the bucket.
+  \param ownerID The canonical ID of the bucket owner.
+   See https://docs.aws.amazon.com/AmazonS3/latest/userguide/finding-canonical-user-id.html for more information.
```

Comment:
For more information, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/finding-canonical-user-id.html.